codalab/worker/file_util.py

```diff
@@ -219,40 +222,64 @@ class OpenFile(object):
     """Opens the file indicated by the given file path and returns a handle
     to the associated file object. Can be in a directory.
 
-    The file path can also refer to a .tar.gz file on Azure.
+    The file path can also refer to an archive file on Blob Storage.
     """
 
     path: str
     mode: str
+    gzipped: bool
 
-    def __init__(self, path: str, mode='r'):
+    def __init__(self, path: str, mode='rb', gzipped=False):
+        """Initialize OpenFile.
+
+        Args:
+            path (str): Path to open; can be a path on disk or a path on Blob Storage.
+            mode (str): Mode with which to open the file. Default is "rb". This is only
+            gzipped (bool): Whether the output should be gzipped. Must be True if downloading a directory;
+                can be True or False if downloading a file.
+        """
         self.path = path
         self.mode = mode
+        self.gzipped = gzipped
 
     def __enter__(self) -> IO[bytes]:
         linked_bundle_path = parse_linked_bundle_url(self.path)
-        if (
-            linked_bundle_path.uses_beam
-            and linked_bundle_path.is_archive
-            and linked_bundle_path.archive_subpath
-        ):
-            # If a file path is specified within a .tar.gz file on Azure, open the specified path within the archive.
-            with OpenIndexedTarGzFile(linked_bundle_path.bundle_path) as tf:
+        if linked_bundle_path.uses_beam and linked_bundle_path.is_archive:
+            # Stream an entire, single .gz file from Blob Storage. This is gzipped by default,
+            # so if the user requested a gzipped version of the entire file, just read and return it.
+            if not linked_bundle_path.is_archive_dir and self.gzipped:
+                return FileSystems.open(self.path, compression_type=CompressionTypes.UNCOMPRESSED)
```

Comment:
we seem to do the same thing whether or not the file is a `.gz` or a `.tar.gz` file (`FileSystems.open(...`)). Would it be possible to condense this logic to make it simpler?

Reply:
No, because we throw an error on the second condition. It seems simpler to just repeat the code because that way the code reads clearer and we can comment out each scenario.

Reply:
fair enough. lgtm then